src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -176,38 +185,64 @@ void GZBridge::clockCallback(const gz::msgs::Clock &msg)
 	px4_clock_settime(CLOCK_MONOTONIC, &ts);
 }
 
-void GZBridge::opticalFlowCallback(const px4::msgs::OpticalFlow &flow)
+void GZBridge::opticalFlowCallback(const px4::msgs::OpticalFlow &msg)
 {
-	sensor_optical_flow_s msg = {};
+	sensor_optical_flow_s report = {};
 
-	msg.timestamp = hrt_absolute_time();
-	msg.timestamp_sample = flow.time_usec();
-	msg.pixel_flow[0] = flow.integrated_x();
-	msg.pixel_flow[1] = flow.integrated_y();
-	msg.quality = flow.quality();
-	msg.integration_timespan_us = flow.integration_time_us();
+	report.timestamp = hrt_absolute_time();
+	report.timestamp_sample = msg.time_usec();
+	report.pixel_flow[0] = msg.integrated_x();
+	report.pixel_flow[1] = msg.integrated_y();
+	report.quality = msg.quality();
+	report.integration_timespan_us = msg.integration_time_us();
 
 	// Static data
 	device::Device::DeviceId id;
 	id.devid_s.bus_type = device::Device::DeviceBusType::DeviceBusType_SIMULATION;
 	id.devid_s.bus = 0;
 	id.devid_s.address = 0;
 	id.devid_s.devtype = DRV_FLOW_DEVTYPE_SIM;
-	msg.device_id = id.devid;
+	report.device_id = id.devid;
 
 	// values taken from PAW3902
-	msg.mode = sensor_optical_flow_s::MODE_LOWLIGHT;
-	msg.max_flow_rate = 7.4f;
-	msg.min_ground_distance = 0.f;
-	msg.max_ground_distance = 30.f;
-	msg.error_count = 0;
+	report.mode = sensor_optical_flow_s::MODE_LOWLIGHT;
+	report.max_flow_rate = 7.4f;
+	report.min_ground_distance = 0.f;
+	report.max_ground_distance = 30.f;
+	report.error_count = 0;
 
 	// No delta angle
 	// No distance
 	// This means that delta angle will come from vehicle gyro
 	// Distance will come from vehicle distance sensor
 
-	_optical_flow_pub.publish(msg);
+	_optical_flow_pub.publish(report);
+}
+
+void GZBridge::magnetometerCallback(const gz::msgs::Magnetometer &msg)
+{
+	const uint64_t timestamp = hrt_absolute_time();
+
+	device::Device::DeviceId id{};
+	id.devid_s.bus_type = device::Device::DeviceBusType::DeviceBusType_SIMULATION;
+	id.devid_s.devtype = DRV_MAG_DEVTYPE_MAGSIM;
+	id.devid_s.bus = 1;
+	id.devid_s.address = 3; // TODO: any value other than 3 causes Commander to not use the mag.... wtf
```

Comment:
@dagar any idea what's going on here? The only way the mag gets used is if this address value is 3. Is this some magic number in Commander for simulated mag to ignore calibration?
```
Preflight Fail: Compass 0 uncalibrated
```